### ISSUE TITLE
Manage UDP packets lost or incomplete

### ISSUE THREAD

**Issue by Tilican** [Windastella/godothub#2]
Hi,
How can we manage UDP packets lost or incomplete ?
For example a join message never get recieved ?

Thx :)

**Comment by Windastella** [comment]
Sadly, there is no handling of packet loss for now. As for now GodotHub is sufficient enough as messaging/chat system. I'm however trying to upgrade it to be suitable for handling multiplayer gameplay by improving performance and stability. The issue of UDP packets lost were usually handled by action prediction system implemented by the game itself, not by network library like GodotHub.

For the example of  join message not received, I assume it was when a player entering a game lobby, yes that is an issue. Probably need to implement received message event or some sort. More overhead will be introduced for stability/integrity tho. But when the client had joined, the game play itself should not have issue with packet loss due to what I had previously explained.

In conclusion, we don't handle packet loss, but we do need to implement a fail-safe in case of join or connected were received or not.

**Comment by Windastella** [comment]
Added Handshake when joining channel when the channel already had members.
